ath79: iodata: extract calibration with nvmem

neheb opened the pull request:
Userspace handling is deprecated.

ping @musashino205 

IIRC the uboot-env driver can't be used because of "" usage.

neheb committed:
ath79: iodata: extract calibration with nvmem

Userspace handling is deprecated.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/18873
Signed-off-by: Robert Marko <robimarko@gmail.com>

musashino205 commented on target/linux/ath79/generic/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac:
```diff
@@ -81,12 +81,13 @@ case "$board" in
 		[ "$PHYNBR" -eq 0 ] && \
 			macaddr_add "$(mtd_get_mac_ascii u-boot-env ethaddr)" 1 > /sys${DEVPATH}/macaddress
 		;;
+		[ "$PHYNBR" -eq 1 ] && \
+			macaddr_add "$(mtd_get_mac_ascii u-boot-env ethaddr)" 0 > /sys${DEVPATH}/macaddress
+		;;
```
Value of `ethaddr` containing `""` is not yet supported.

neheb committed:
ath79: iodata: set ath9k LED pin

The ath9k driver creates an ath9k LED by default. Instead of having a
non functional LED, configure it properly and remove the extra as it's
not needed.

Set the pcie device to qcom,ath10k as the Documentation says, The pci
line implies it's ath9k when it's actually ath10k.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/18873
Signed-off-by: Robert Marko <robimarko@gmail.com>

neheb commented:
@musashino205 fixed a small documentation bug and added led-sources support.

neheb commented on target/linux/ath79/dts/ar1022_iodata_wn-ag300dgr.dts:
```diff
@@ -207,5 +211,11 @@
+	nvmem-cells = <&cal_art_1000>;
+	nvmem-cell-names = "calibration";
+
+	led {
+		led-sources = <12>;
+		led-active-low;
+	};
```
AFAIK for the wmac, the gpio controller is the same. For pcie it’s different
